bpfd-api/src/config.rs

```diff
@@ -95,6 +97,26 @@ impl ToString for XdpMode {
     }
 }
 
+#[derive(Debug, Deserialize)]
+pub struct Grpc {
+    pub endpoint: Endpoint,
+}
+
+impl Default for Grpc {
+    fn default() -> Self {
+        Self {
+            endpoint: Endpoint {
+                address: String::from("[::1]:50051"),
+            },
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Endpoint {
+    pub address: String,
```

Comment:
I think this should be two fields, an address and a port. If you search through the code, there are several places that use the hard coded port 50051 (XDP example, TC example and bpfd-agent). They all use:
`grpc.DialContext(ctx, "localhost:50051", grpc.WithTransportCredentials(creds))`
I think this code will want to dial "localhost:"{port}. All these instances currently read bpfd.toml to get the certs, so they can get the port as well.

Reply:
I split the fields, do you want me to update those other places that are using the hardcoded port as part of this PR? Or should that be addressed in a follow up?

Reply:
I'd say do it now if it's not too much trouble, because if the port is changed, all the hardcoded places break. But it you want to do it as a follow-up, it needs to come closely after.

Reply:
Changed the examples and the agent to use the port from the configuration, I'm terrible at go, so I apologize in advance.